eldritch/test_unique_items.py

```diff
@@ -0,0 +1,492 @@
+    self.state.places["Woods"].gate = self.get_gate("Sunken City")
+    self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Woods")
+
+  def testSaveLostInTimeAndSpace(self):
+    self.state.places["Woods"].gate = self.get_gate("Abyss")
+    self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Woods")
+
+
+class ObsidianStatueTest(EventTest):
+  def setUp(self):
+    super().setUp()
+    self.statue = items.ObsidianStatue(0)
+    self.char.possessions = [self.statue]
+
+  def testClueLoss(self):
+    self.state.event_stack.append(events.Loss(self.char, {"clues": 1}))
+    with self.assertRaises(AssertionError):
+      self.resolve_to_usable(0, "Obsidian Statue0")
+
+  def testSingleSanity(self):
+    self.state.event_stack.append(events.Loss(self.char, {"sanity": 1}))
+    statue = self.resolve_to_usable(0, "Obsidian Statue0")
+    self.state.event_stack.append(statue)
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 5)
+    self.assertNotIn(self.statue, self.char.possessions)
+
+  def testMultipleSanity(self):
+    self.state.event_stack.append(events.Loss(self.char, {"sanity": 2}))
+    statue = self.resolve_to_usable(0, "Obsidian Statue0")
+    self.state.event_stack.append(statue)
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 5)
+    self.assertNotIn(self.statue, self.char.possessions)
+
+  def testSingleStamina(self):
+    self.state.event_stack.append(events.Loss(self.char, {"stamina": 1}))
+    statue = self.resolve_to_usable(0, "Obsidian Statue0")
+    self.state.event_stack.append(statue)
+    self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 5)
+    self.assertNotIn(self.statue, self.char.possessions)
+
+  def testMultipleStamina(self):
+    self.state.event_stack.append(events.Loss(self.char, {"stamina": 2}))
+    statue = self.resolve_to_usable(0, "Obsidian Statue0")
+    self.state.event_stack.append(statue)
+    self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 5)
+    self.assertNotIn(self.statue, self.char.possessions)
+
+  def testSanityAndStamina(self):
+    self.state.event_stack.append(events.Loss(self.char, {"stamina": 2, "sanity": 2}))
+    statue = self.resolve_to_usable(0, "Obsidian Statue0")
+    self.state.event_stack.append(statue)
+    choice = self.resolve_to_choice(events.MultipleChoice)
+    choice.resolve(self.state, "Stamina")
+    self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 5)
+    self.assertEqual(self.char.sanity, 3)
+    self.assertNotIn(self.statue, self.char.possessions)
+
+
+class SilverKeyTest(EventTest):
+  def setUp(self):
+    super().setUp()
+    self.key = items.SilverKey(0)
+    self.char.possessions.append(self.key)
+    combat = events.Combat(
+        self.char, monsters.DreamFlier()
+    )
+    self.state.event_stack.append(combat)
+    evade_choice = self.resolve_to_choice(events.FightOrEvadeChoice)
+    evade_choice.resolve(self.state, "Flee")
+
```

Comment:
Can you also add a test that validates that you can use the silver key on the initial evade check that you make for an EvadeOrCombat? And also validate that if you use the silver key before entering combat that your movement does not end?